### DISCUSSION TITLE
Support for DisplayLink (USB display adapter or dock)

### DISCUSSION THREAD

**Discussion by GeoffreyPlitt** [MonitorControl/MonitorControl#583] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] I have read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

I have an HDMI monitor (Dell S2315H) connected via a USB dongle (https://www.amazon.com/gp/product/B01CSG7TUC/)

This software works great for my other monitors (USB-C or HDMI direct), but not for this one connected via USB adapter.

In preferences, it shows it is aware of the monitor, but says "No Control Available"

Wondering if it is possible fix?

### Expected behavior

Dimming for this monitor.

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.2.3 Bug Sur
- Mac model: Mac Mini M1/2020
- MonitorControl version: 3.0.0 Build 3097
- Monitor(s): Dell S2315H
- Monitor Cable(s)/Connection(s):  https://www.amazon.com/gp/product/B01CSG7TUC and then HDMI
- Apple Silicon/M1 (yes or no): yes
```

**Comment by waydabber** [comment, marked as answer]
Hi @GeoffreyPlitt 

DisplayLink currently does not support DDC/CI on macOS so there is no way to add support for these devices.

You can raise your voice regarding this here: https://support.displaylink.com/forums/287786-displaylink-feature-suggestions/suggestions/16135240-add-brightness-support-to-non-windows-oses

Also, as far as I know DisplayLink drivers are incompatible with gamma table calibration (used for software dimming), but I am not entirely sure about this, more experimentation would be needed with this. MonitorControl currently disables any kind of control for DisplayLink devices.

I convert this issue to a discussion, maybe the situation will change.

**Reply by GeoffreyPlitt** [reply]
Thanks, I'll post in that forum!

**Reply by waydabber** [reply]
3.1.0 will some limited software based support for DisplayLink as well as virtual displays (AirPlay, screen sharing, iPad) by using window shades. This is inferior than hardware control and even gamma based control but still better than nothing.

**Reply by Tianhebeilu** [reply]
Hi, I downloaded the 3.1.1 but still not working with my DisplayLink monitor.
I cannot use the "overlay" dimmer. Brightness does not change at all.
Did I miss anything?
Thanks

**Reply by waydabber** [reply]
Hey, sorry what we planned to be 3.1.0 actually became 4.0.0. 3.1.x were just service releases with minor new features. Download the new v4.0.0 please!

**Reply by waydabber** [reply]
See this: https://github.com/MonitorControl/MonitorControl/releases/tag/v4.0.0-rc1

**Comment by Tianhebeilu** [comment]
I cannot find your e-mail.
Can you please share it with me?
Best

**Reply by waydabber** [reply]
Hmm. Which email? :) I mean you should download version 4.0.0 from the following link for limited DisplayLink support:

https://github.com/MonitorControl/MonitorControl/releases/tag/v4.0.0-rc1

**Comment by Tianhebeilu** [comment]
Haha I know, thanks.
But I meant your own personal e-mail.
As I got a couple of questions to ask you without bothering the whole
community.
Thanks :)

**Reply by waydabber** [reply]
It's ok I think, these discussions are not widely read and the discussion might be educational to all.
